src/main/java/io/split/android/client/attributes/AttributesManagerImpl.java

```diff
@@ -40,7 +55,11 @@ public boolean setAttribute(String attributeName, Object value) {
             return false;
         }
 
-        mSplitTaskExecutor.submit(mSplitTaskFactory.createUpdateSingleAttributeTask(attributeName, value), null);
+        mAttributesStorage.set(attributeName, value);
+
+        if (mPersistentAttributesStorage != null) {
```

Comment:
It would be good to remove this  nullity check from every function in favor of one check in _**submitUpdateTask**_ to avoid code duplication.